engine/runtime/src/main/scala/org/enso/interpreter/instrument/execution/JobExecutionEngine.scala

```diff
@@ -31,17 +31,26 @@ class JobExecutionEngine(
   private val runningJobsRef =
     new AtomicReference[Vector[RunningJob]](Vector.empty)
 
+  private val backgroundJobsRef =
+    new AtomicReference[Vector[RunningJob]](Vector.empty)
+
   private val context = interpreterContext.executionService.getContext
 
   private val jobParallelism =
     interpreterContext.executionService.getContext.getEnvironment.getOptions
       .get(RuntimeServerInfo.JOB_PARALLELISM_KEY)
       .intValue()
 
-  val jobExecutor = Executors.newFixedThreadPool(
-    jobParallelism,
-    new TruffleThreadFactory(context, "job-pool")
-  )
+  val jobExecutor: ExecutorService =
+    Executors.newFixedThreadPool(
+      jobParallelism,
+      new TruffleThreadFactory(context, "job-pool")
+    )
+
+  val backgroundJobExecutor: ExecutorService =
+    Executors.newSingleThreadExecutor(
+      new TruffleThreadFactory(context, "background-job-pool")
```

Comment:
The thread name will include `*pool`, but as we are creating `newSingleThreadExecutor`, it is not going to be really a _pool_ ;-)

How is the parallelism supposed to work? There is `jobExecutor` pool containing `JOB_PARALLELISM_KEY` threads. All of them are going to share this single background thread. Is there any rationale behind this? I mean: separating _urgent work_ and _pending work_ is good, just the ratio between the _pools_ puzzles me.

In the context of [too many threads problem](https://www.pivotaltracker.com/n/projects/2539304/stories/182283991): smaller pools might be more suitable for _one client/one server_ communication.

Reply:
The separate executor is required to run the background jobs independently of the main program flow, mainly the compilation and execution. And for now we want the background jobs to be sequential, to keep the same semantics as before. We can relax this restriction later